shared/src/ledger/ibc/handler.rs

```diff
@@ -1111,13 +1111,19 @@ pub fn packet_from_message(
 }
 
 /// Returns a commitment from the given packet
-pub fn commitment(packet: &Packet) -> String {
-    let input = format!(
-        "{:?},{:?},{:?}",
-        packet.timeout_timestamp, packet.timeout_height, packet.data,
-    );
-    let r = sha2::Sha256::digest(input.as_bytes());
-    format!("{:x}", r)
+pub fn commitment(packet: &Packet) -> PacketCommitment {
+    let mut input = packet
+        .timeout_timestamp
+        .nanoseconds()
+        .to_be_bytes()
+        .to_vec();
+    let revision_number = packet.timeout_height.revision_number.to_be_bytes();
+    input.append(&mut revision_number.to_vec());
+    let revision_height = packet.timeout_height.revision_height.to_be_bytes();
+    input.append(&mut revision_height.to_vec());
+    let data = sha2::Sha256::digest(&packet.data);
```

Comment:
Just checking here, is the `packet.data` meant to be hashed on its own first?

Unrelated nit, `extend_from_slice` is cheaper than `append`, but slice concat function is better still as it allocates only once for the total size

Reply:
Yes, the `packet.data` is hashed first, then appended.
Ref. https://github.com/cosmos/ibc-go/blob/c86d27fc280cfb342a9e4689b381e5823441b694/modules/core/04-channel/types/packet.go#L19